data/lua/core/unit_test.lua

```diff
@@ -44,6 +44,31 @@ if rawget(_G, 'unit_test') ~= nil then
 		std_print(prefix .. ': ' .. message)
 	end
 
+	-- This is a way to ensure that all assertions contain a descriptive message
+	setmetatable(unit_test, {
+		__newindex = function(self, key, val)
+			std_print('adding unit_test.' .. key)
+			if string.sub(key, 1, 6) == 'assert' then
```

Comment:
Seems overly complex and also prone to false positives (although I haven't built your branch to test this). What if a unit test defines a local `assert_foo(a, expectation)` where the message string is inside `assert_foo` itself?

Reply:
I think that's a _relatively_ unlikely case, but they can always use `rawset` for it if they have to.